frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/projects/ProjectPortlet.java

```diff
@@ -1001,6 +1045,31 @@ private void prepareLicenseInfo(RenderRequest request, RenderResponse response)
         }
     }
 
+    private Set<String> fetchReleaseRelationships(List<ProjectLink> mappedProjectLinks) {
+        return mappedProjectLinks.stream().map(ProjectLink::getLinkedReleases).flatMap(List::stream)
+                .map(ReleaseLink::getReleaseRelationship).map(ThriftEnumUtils::enumToString)
+                .collect(Collectors.toSet());
+    }
+
+    private Set<String> fetchIcludedReleaseRelationships(String projectId) {
+        Set<String> excludedRelationships = Sets.newHashSet();
+        try {
+            final AttachmentService.Iface attachmentClient = thriftClients.makeAttachmentClient();
+            List<AttachmentUsage> usages = attachmentClient.getUsedAttachments(Source.projectId(projectId), null);
+            excludedRelationships = usages.stream().map(AttachmentUsage::getUsageData).filter(Objects::nonNull)
+                    .filter(ud -> ud.getSetField().equals(UsageData._Fields.LICENSE_INFO))
+                    .map(UsageData::getLicenseInfo).filter(Objects::nonNull)
+                    .filter(LicenseInfoUsage::isSetIncludedReleaseRelationship)
+                    .map(LicenseInfoUsage::getIncludedReleaseRelationship).filter(Objects::nonNull)
+                    .flatMap(Collection::stream).collect(Collectors.toSet());
+        } catch (TException exception) {
+            log.error("cannot retrieve information about attachment usages.", exception);
+        }
+
+        return excludedRelationships;
+
```

Comment:
Please remove extra line break from end of function

Reply:
Done